### PR TITLE
cri-o, buildah: add wrappers

### PR DESCRIPTION
**Pull request by zowoq**
~~Opened as a draft, waiting for other changes to land in master.~~

~~Having a wrapper might be preferred but for now I'd rather that we merge this and reevaluate if we end up adding more packages.~~

Ended up being easier to add the wrapper.

cc @NixOS/podman

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
LGTM

**Comment by zowoq** [comment]
Needs to be rebased after https://github.com/NixOS/nixpkgs/pull/86290 is merged.

File: nixos/modules/virtualisation/cri-o.nix [hunk 1]
```diff
@@ -23,55 +25,90 @@ in
...
-      pause_command = "${cfg.pauseCommand}"
+      ${optionalString (cfg.pauseImage != null) ''pause_image = "${cfg.pauseImage}"''}
+      ${optionalString (cfg.pauseCommand != null) ''pause_command = "${cfg.pauseCommand}"''}
 
       [crio.network]
       plugin_dirs = ["${pkgs.cni-plugins}/bin/"]
-      network_dir = "/etc/cni/net.d/"
 
       [crio.runtime]
-      conmon = "${pkgs.conmon}/bin/conmon"
```
Comment (zowoq) [inline nixos/modules/virtualisation/cri-o.nix:69]: @saschagrunert Apologies for asking you to keep this in #86290 and then removing it here anyway, the wrapper end up being the best approach.

Comment (saschagrunert) [reply nixos/modules/virtualisation/cri-o.nix:69]: Yeah no worries :heart:

**Comment by Profpatsch** [comment]
I merged https://github.com/NixOS/nixpkgs/pull/86290 so this should be good to go?

**Comment by zowoq** [comment]
> I merged #86290 so this should be good to go?

Yes, thanks @Profpatsch

File: pkgs/development/tools/buildah/wrapper.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    makeWrapper
+  ];
+
+} ''
+  ln -s ${buildah.man} $man
+
+  mkdir -p $out/bin
+  ln -s ${buildah-unwrapped}/share $out/share
+  makeWrapper ${buildah-unwrapped}/bin/buildah $out/bin/buildah \
+    --prefix PATH : ${binPath}
```
Comment (Profpatsch) [inline pkgs/development/tools/buildah/wrapper.nix:51]: iirc you don’t need a separate derivation to wrap the executable, you can just run `makeWrapper` in the `postInstall` phase of the original derivation.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -1369,7 +1369,8 @@ in
 
   btfs = callPackage ../os-specific/linux/btfs { };
 
-  buildah = callPackage ../development/tools/buildah { };
+  buildah = callPackage ../development/tools/buildah/wrapper.nix { };
+  buildah-unwrapped = callPackage ../development/tools/buildah { };
```
Comment (Profpatsch) [inline pkgs/top-level/all-packages.nix:1373]: Do we need to expose the unwrapped version?

File: pkgs/development/tools/buildah/wrapper.nix (hunk 2 above)
Comment (zowoq) [reply pkgs/development/tools/buildah/wrapper.nix:51]: This and the top-level `-unwrapped` are duplicated from `podman`.

For now I'd like to keep them as duplicates so they are easier to merge into one wrapper. I'm looking at doing that after adding some tests.

File: pkgs/applications/virtualization/cri-o/wrapper.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+, lib
+, extraPackages ? []
+, cri-o
+, runc # Default container runtime
+, crun # Container runtime (default with cgroups v2 for podman/buildah)
+, conmon # Container runtime monitor
+, utillinux # nsenter
+, cni-plugins # not added to path
+, iptables
+}:
```
Comment (saschagrunert) [inline pkgs/applications/virtualization/cri-o/wrapper.nix:13]: We still need `socat`for port forwarding purposes.

**Review by saschagrunert — APPROVED** [review]

File: pkgs/applications/virtualization/cri-o/wrapper.nix (hunk 4 above)
Comment (zowoq) [reply pkgs/applications/virtualization/cri-o/wrapper.nix:14]: Fixed.

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**